### PR TITLE
Feat/update data process

### LINKED ISSUE #778: [Bug] scatter size is not correct after updateSpec
### Version

1.3.0

### Link to Minimal Reproduction

null

### Steps to Reproduce

call updateSpec with the same spec config：
```typescript
const spec = {
    "type": "scatter",
    "xField": "230904155356016",
    "yField": "230904155356019",
    "padding": 0,
    "data": [
        {
            "id": "data",
            "values": [
                {
                    "30001": "椅子",
                    "230904155356016": "1949.4720458984375",
                    "230904155356019": "0.10000000149011612",
                    "230904155356025": "8",
                    "230904155524028": "驻马店",
                    "230904160355011": "椅子"
                },
                {
                    "30001": "书架",
                    "230904155356016": "581.4199829101562",
                    "230904155356019": "0",
                    "230904155356025": "1",
                    "230904155524028": "莱阳",
                    "230904160355011": "书架"
                },
                {
                    "30001": "桌子",
                    "230904155356016": "1814.4000244140625",
                    "230904155356019": "0.25",
                    "230904155356025": "1",
                    "230904155524028": "佳木斯",
                    "230904160355011": "桌子"
                },
                {
                    "30001": "系固件",
                    "230904155356016": "372.3999938964844",
                    "230904155356019": "0",
                    "230904155356025": "7",
                    "230904155524028": "宣州",
                    "230904160355011": "系固件"
                },
                {
                    "30001": "器具",
                    "230904155356016": "4081.139892578125",
                    "230904155356019": "0",
                    "230904155356025": "9",
                    "230904155524028": "滴道",
                    "230904160355011": "器具"
                },
                {
                    "30001": "用具",
                    "230904155356016": "2039.6600341796875",
                    "230904155356019": "0",
                    "230904155356025": "4",
                    "230904155524028": "秦皇岛",
                    "230904160355011": "用具"
                },

            ],
            "fields": {
                "30001": {
                    "alias": "图例项 "
                },
                "230904155356016": {
                    "alias": "销售额"
                },
                "230904155356019": {
                    "alias": "折扣"
                },
                "230904155356025": {
                    "alias": "数量"
                },
                "230904155524028": {
                    "alias": "城市",
                    "lockStatisticsByDomain": true
                },
                "230904160355011": {
                    "alias": "子类别"
                }
            }
        }
    ],
    "axes": [
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#8D8D8D"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#DADCDD",
                    "lineWidth": 1,
                    "lineDash": [
                        4,
                        2
                    ]
                }
            },
            "orient": "left",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#8D8D8D"
                }
            },
            "title": {
                "visible": false,
                "text": "折扣",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "sampling": false,
            "label": {
                "visible": true,
                "space": 8,
                "flush": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal"
                },
                "autoHide": true,
                "autoHideMethod": "greedy"
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        },
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#8D8D8D"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#DADCDD",
                    "lineWidth": 1,
                    "lineDash": [
                        4,
                        2
                    ]
                }
            },
            "orient": "bottom",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#8D8D8D"
                }
            },
            "title": {
                "visible": false,
                "text": "销售额",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "sampling": false,
            "label": {
                "visible": true,
                "space": 4,
                "flush": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal"
                },
                "autoHide": true,
                "autoHideMethod": "greedy"
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        }
    ],
    "label": {
        "visible": false,
        "overlap": {
            "hideOnHit": true,
            "clampForce": true
        },
        "style": {
            "fontSize": 12,
            "fontWeight": "normal",
            "fill": "#363839",
            "stroke": "rgba(255, 255, 255, 0.8)",
            "lineWidth": 2,
            "strokeOpacity": 1
        },
        "offset": 6,
        "position": "top"
    },
    "legends": [
        {
            "type": "discrete",
            "id": "legend",
            "orient": "right",
            "position": "start",
            "layoutType": "normal-inline",
            "visible": true,
            "maxCol": 1,
            "title": {
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "layoutLevel": 60,
            "maxHeight": 530,
            "item": {
                "focus": true,
                "focusIconStyle": {
                    "size": 14
                },
                "maxWidth": 438,
                "spaceRow": 0,
                "spaceCol": 0,
                "padding": {
                    "top": 1,
                    "bottom": 2,
                    "left": 3,
                    "right": 2
                },
                "background": {
                    "visible": false,
                    "style": {
                        "fillOpacity": 0.001
                    }
                },
                "label": {
                    "style": {
                        "fontSize": 12,
                        "fill": "#6F6F6F"
                    }
                },
                "shape": {
                    "style": {
                        "lineWidth": 0,
                        "symbolType": "square"
                    }
                }
            },
            "pager": {
                "padding": {
                    "top": -10,
                    "left": -18
                },
                "textStyle": {},
                "space": 0,
                "handler": {
                    "preShape": "triangleLeft",
                    "nextShape": "triangleRight",
                    "style": {},
                    "state": {
                        "disable": {}
                    }
                }
            },
            "padding": {
                "top": 0,
                "bottom": 12,
                "left": 16,
                "right": 0
            }
        },
        {
            "type": "size",
            "slidable": true,
            "title": {
                "visible": true,
                "text": "数量",
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                },
                "space": 0
            },
            "id": "legend",
            "orient": "right",
            "position": "start",
            "layoutType": "normal-inline",
            "visible": true,
            "field": [
                "230904155356025"
            ],
            "layoutLevel": 50,
            "maxHeight": 130,
            "width": 100,
            "track": {
                "style": {
                    "fill": "rgba(20,20,20,0.1)"
                }
            },
            "rail": {
                "height": 100
            },
            "handler": {
                "style": {
                    "symbolType": "M-5,0L-2.5,-3.5\n    h5\n    v7\n    h-5\n    Z",
                    "size": 8,
                    "outerBorder": {
                        "stroke": "#AEB8C6",
                        "lineWidth": 1,
                        "strokeOpacity": 0.5
                    }
                }
            },
            "handlerText": {
                "style": {
                    "maxLineWidth": 50,
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "padding": {
                "top": 0,
                "bottom": 12,
                "left": 23,
                "right": 0
            }
        }
    ],
    "seriesField": "230904155524028",
    "color": {
        "field": "230904155524028",
        "type": "ordinal",

        "specified": {},

    },
    "sizeField": "230904155356025",
    "size": {
        "type": "linear",
        "range": [
            7.0898154036220635,
            28.359261614488254
        ]
    },
    "background": "rgba(255, 255, 255, 0)",
    "point": {
        "style": {
            "fillOpacity": 1
        },
        "state": {
            "hover": {
                "cursor": "pointer",
                "fillOpacity": 0.8,
                "stroke": "#58595B",
                "lineWidth": 1,
                "zIndex": 500
            }
        }
    },
    "seriesMark": "point",
    "hover": {
        "enable": true
    },
    "select": {
        "enable": true
    },
    "animation": false,
    "hash": "88ac5746ee5611effa05ebc368b948f1"
}

const spec2={
    "type": "scatter",
    "xField": "230904155356016",
    "yField": "230904155356019",
    "padding": 0,
    "data": [
        {
            "id": "data",
            "values": [
                {
                    "30001": "椅子",
                    "230904155356016": "1949.4720458984375",
                    "230904155356019": "0.10000000149011612",
                    "230904155356025": "8",
                    "230904155524028": "驻马店",
                    "230904160355011": "椅子"
                },
                {
                    "30001": "书架",
                    "230904155356016": "581.4199829101562",
                    "230904155356019": "0",
                    "230904155356025": "1",
                    "230904155524028": "莱阳",
                    "230904160355011": "书架"
                },
                {
                    "30001": "桌子",
                    "230904155356016": "1814.4000244140625",
                    "230904155356019": "0.25",
                    "230904155356025": "1",
                    "230904155524028": "佳木斯",
                    "230904160355011": "桌子"
                },
                {
                    "30001": "系固件",
                    "230904155356016": "372.3999938964844",
                    "230904155356019": "0",
                    "230904155356025": "7",
                    "230904155524028": "宣州",
                    "230904160355011": "系固件"
                },
                {
                    "30001": "器具",
                    "230904155356016": "4081.139892578125",
                    "230904155356019": "0",
                    "230904155356025": "9",
                    "230904155524028": "滴道",
                    "230904160355011": "器具"
                },
                {
                    "30001": "用具",
                    "230904155356016": "2039.6600341796875",
                    "230904155356019": "0",
                    "230904155356025": "4",
                    "230904155524028": "秦皇岛",
                    "230904160355011": "用具"
                },

            ],
            "fields": {
                "30001": {
                    "alias": "图例项 "
                },
                "230904155356016": {
                    "alias": "销售额"
                },
                "230904155356019": {
                    "alias": "折扣"
                },
                "230904155356025": {
                    "alias": "数量"
                },
                "230904155524028": {
                    "alias": "城市",
                    "lockStatisticsByDomain": true
                },
                "230904160355011": {
                    "alias": "子类别"
                }
            }
        }
    ],
    "axes": [
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#8D8D8D"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#DADCDD",
                    "lineWidth": 1,
                    "lineDash": [
                        4,
                        2
                    ]
                }
            },
            "orient": "left",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#8D8D8D"
                }
            },
            "title": {
                "visible": false,
                "text": "折扣",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "sampling": false,
            "label": {
                "visible": true,
                "space": 8,
                "flush": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal"
                },
                "autoHide": true,
                "autoHideMethod": "greedy"
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        },
        {
            "type": "linear",
            "tick": {
                "visible": false,
                "style": {
                    "stroke": "#8D8D8D"
                }
            },
            "niceType": "accurateFirst",
            "grid": {
                "visible": true,
                "style": {
                    "stroke": "#DADCDD",
                    "lineWidth": 1,
                    "lineDash": [
                        4,
                        2
                    ]
                }
            },
            "orient": "bottom",
            "visible": true,
            "domainLine": {
                "visible": true,
                "style": {
                    "lineWidth": 1,
                    "stroke": "#8D8D8D"
                }
            },
            "title": {
                "visible": false,
                "text": "销售额",
                "style": {
                    "fontSize": 12,
                    "fill": "#363839",
                    "fontWeight": "normal"
                }
            },
            "sampling": false,
            "label": {
                "visible": true,
                "space": 4,
                "flush": true,
                "style": {
                    "fontSize": 12,
                    "fill": "#6F6F6F",
                    "angle": 0,
                    "fontWeight": "normal"
                },
                "autoHide": true,
                "autoHideMethod": "greedy"
            },
            "hover": true,
            "background": {
                "visible": true,
                "state": {
                    "hover": {
                        "fillOpacity": 0.08,
                        "fill": "#141414"
                    }
                }
            },
            "zero": true,
            "nice": true
        }
    ],
    "label": {
        "visible": false,
        "overlap": {
            "hideOnHit": true,
            "clampForce": true
        },
        "style": {
            "fontSize": 12,
            "fontWeight": "normal",
            "fill": "#363839",
            "stroke": "rgba(255, 255, 255, 0.8)",
            "lineWidth": 2,
            "strokeOpacity": 1
        },
        "offset": 6,
        "position": "top"
    },
    "legends": [
        {
            "type": "discrete",
            "id": "legend",
            "orient": "right",
            "position": "start",
            "layoutType": "normal-inline",
            "visible": true,
            "maxCol": 1,
            "title": {
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "layoutLevel": 60,
            "maxHeight": 530,
            "item": {
                "focus": true,
                "focusIconStyle": {
                    "size": 14
                },
                "maxWidth": 438,
                "spaceRow": 0,
                "spaceCol": 0,
                "padding": {
                    "top": 1,
                    "bottom": 2,
                    "left": 3,
                    "right": 2
                },
                "background": {
                    "visible": false,
                    "style": {
                        "fillOpacity": 0.001
                    }
                },
                "label": {
                    "style": {
                        "fontSize": 12,
                        "fill": "#6F6F6F"
                    }
                },
                "shape": {
                    "style": {
                        "lineWidth": 0,
                        "symbolType": "square"
                    }
                }
            },
            "pager": {
                "padding": {
                    "top": -10,
                    "left": -18
                },
                "textStyle": {},
                "space": 0,
                "handler": {
                    "preShape": "triangleLeft",
                    "nextShape": "triangleRight",
                    "style": {},
                    "state": {
                        "disable": {}
                    }
                }
            },
            "padding": {
                "top": 0,
                "bottom": 12,
                "left": 16,
                "right": 0
            }
        },
        {
            "type": "size",
            "slidable": true,
            "title": {
                "visible": true,
                "text": "数量",
                "textStyle": {
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                },
                "space": 0
            },
            "id": "legend",
            "orient": "right",
            "position": "start",
            "layoutType": "normal-inline",
            "visible": true,
            "field": [
                "230904155356025"
            ],
            "layoutLevel": 50,
            "maxHeight": 130,
            "width": 100,
            "track": {
                "style": {
                    "fill": "rgba(20,20,20,0.1)"
                }
            },
            "rail": {
                "height": 100
            },
            "handler": {
                "style": {
                    "symbolType": "M-5,0L-2.5,-3.5\n    h5\n    v7\n    h-5\n    Z",
                    "size": 8,
                    "outerBorder": {
                        "stroke": "#AEB8C6",
                        "lineWidth": 1,
                        "strokeOpacity": 0.5
                    }
                }
            },
            "handlerText": {
                "style": {
                    "maxLineWidth": 50,
                    "fontSize": 12,
                    "fill": "#6F6F6F"
                }
            },
            "padding": {
                "top": 0,
                "bottom": 12,
                "left": 23,
                "right": 0
            }
        }
    ],
    "seriesField": "230904155524028",
    "color": {
        "field": "230904155524028",
        "type": "ordinal",

        "specified": {},

    },
    "sizeField": "230904155356025",
    "size": {
        "type": "linear",
        "range": [
            7.0898154036220635,
            28.359261614488254
        ]
    },
    "background": "rgba(255, 255, 255, 0)",
    "point": {
        "style": {
            "fillOpacity": 1
        },
        "state": {
            "hover": {
                "cursor": "pointer",
                "fillOpacity": 0.8,
                "stroke": "#58595B",
                "lineWidth": 1,
                "zIndex": 500
            }
        }
    },
    "seriesMark": "point",
    "hover": {
        "enable": true
    },
    "select": {
        "enable": true
    },
    "animation": false,
    "hash": "88ac5746ee5611effa05ebc368b948f1"
}

  const cs = new VChart(spec, {
    dataSet,
    dom: document.getElementById('chart') as HTMLElement,
    mode: isMobile ? 'mobile-browser' : 'desktop-browser'
  });

  setTimeout(()=>{
    cs.updateSpec(spec2)
  },1000)

```

### Current Behavior

image

### Expected Behavior

image

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by xuefei1313**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #778 #912 #1007

### 🐞 Bugserver case id

650bb83fcc0f2bdafa17759c
650bb8830927bc458fdcef50
650bb9100927bc458fdcef51

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/component/axis/cartesian/axis.ts [hunk 1]
```diff
@@ -111,6 +111,13 @@ export abstract class CartesianAxis<T extends ICartesianAxisCommonSpec = ICartes
     this._dataSet = options.dataSet;
   }
 
+  static transformInverse(spec: any, isHorizontal: boolean) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/axis/cartesian/axis.ts:114]: 这个抽成一个工具方法会更合适一些？

File: packages/vchart/src/chart/chart-meta/data.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return;
+      }
+      callBack(data, dv, i);
+    });
+  }
+
+  getSeriesData(id: StringOrNumber | undefined, index: number | undefined): DataView | undefined {
+    if (!this._dataArr.length) {
+      // 没有数据，报错处理
+      this._onError?.('no data in spec!');
```
Comment (xiaoluoHe) [inline packages/vchart/src/chart/chart-meta/data.ts:63]: 地图有可能没有系列数据的，这里会直接报错吗？

**Comment by xiaoluoHe** [comment]
https://github.com/VisActor/VChart/blob/c0ccc60b9e5ff47c64a798c0517bf66dc7a144c4/packages/vchart/src/model/interface.ts#L208

`reMakeData`是不是可以删掉了？

File: packages/vchart/src/component/axis/cartesian/axis.ts (hunk 1 above)
Comment (xuefei1313) [reply packages/vchart/src/component/axis/cartesian/axis.ts]: done

File: packages/vchart/src/chart/chart-meta/data.ts (hunk 2 above)
Comment (xuefei1313) [reply packages/vchart/src/chart/chart-meta/data.ts:63]: 额，地图没有系列数据，应该也不会调用这个方法？我试试

File: packages/vchart/src/compile/compiler.ts [hunk 3]
```diff
@@ -257,7 +257,7 @@ export class Compiler {
     this._view.renderer.setViewBox(viewBox, reRender);
   }
 
-  resize(width: number, height: number) {
+  resize(width: number, height: number, reRender: boolean = true) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/compile/compiler.ts:260]: 这个 reRender好像没啥用？

File: packages/vchart/src/component/base/base-component.ts [hunk 4]
```diff
@@ -121,6 +125,7 @@ export abstract class BaseComponent<T extends IComponentSpec = IComponentSpec>
    * updateSpec
    */
   updateSpec(spec: any) {
+    this._mergeThemeToSpec;
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/base/base-component.ts:128]: 这个没有调用也

File: packages/vchart/src/compile/compiler.ts (hunk 3 above)
Comment (xuefei1313) [reply packages/vchart/src/compile/compiler.ts:260]: 是的，我删掉

File: packages/vchart/src/component/base/base-component.ts (hunk 4 above)
Comment (xuefei1313) [reply packages/vchart/src/component/base/base-component.ts:128]: 额， 删了。。

**Comment by xuefei1313** [comment]
> https://github.com/VisActor/VChart/blob/c0ccc60b9e5ff47c64a798c0517bf66dc7a144c4/packages/vchart/src/model/interface.ts#L208
> 
> `reMakeData`是不是可以删掉了？

done

**Pull request merged**